### PR TITLE
Jetpack Cloud error pages

### PR DESCRIPTION
**Pull request by ChaosExAnima**
#### Changes proposed in this Pull Request

* This adds logic for 404 and 500 error pages for Jetpack Cloud. These pages are loaded using the same component as Calypso pages, but load the Jetpack Cloud CSS instead.

#### Testing instructions

* Build Jetpack Cloud via `CALYPSO_ENV=jetpack-cloud-development npm start`.
* Go to route that doesn't exist.
* Observe error page.
* Build Calypso and smoke test.

Fixes 1156570014567299-as-1159128713000441

### CODE REVIEW

File: client/server/pages/index.js [hunk 1]
```diff
@@ -22,7 +22,8 @@ import {
 	split,
 } from 'lodash';
 import bodyParser from 'body-parser';
-import superagent from 'superagent';
+// eslint-disable-next-line no-restricted-imports
+import superagent from 'superagent'; // Don't have Node.js fetch lib yet.
```
Comment (ChaosExAnima) [inline client/server/pages/index.js]: Raised #39115 to resolve this.

File: client/landing/jetpack-cloud/layout.js [hunk 2]
```diff
@@ -25,7 +26,7 @@ class JetpackCloudLayout extends Component {
 						{ this.props.secondary }
 					</div>
 					<div id="primary" className="layout__primary">
-						{ this.props.primary }
+						{ this.props.primary || <EmptyContent /> }
```
Comment (tyxla) [inline client/landing/jetpack-cloud/layout.js]: This is a bit weird. Could we set `context.primary` to an `<EmptyContent />` earlier if we're running into a 404?

File: client/server/pages/index.js [hunk 3]
```diff
@@ -655,6 +658,13 @@ module.exports = function() {
 		JETPACK_CLOUD_SECTION_DEFINITION.paths.forEach( sectionPath =>
 			handleSectionPath( JETPACK_CLOUD_SECTION_DEFINITION, sectionPath, 'entry-jetpack-cloud' )
 		);
+
+		// catchall to render 404 for all routes not whitelisted in client/sections
+		app.use( render404( 'entry-jetpack-cloud' ) );
```
Comment (tyxla) [inline client/server/pages/index.js]: Seems like the `render404` and `renderServerError` middlewares go together both here and when hooking to the main Calypso app. Should we abstract that in a helper for that to reduce repetition?

File: client/landing/jetpack-cloud/layout.js (hunk 2 above)
Comment (ChaosExAnima) [reply client/landing/jetpack-cloud/layout.js]: This code pathway isn't actually ever called if we hit a 404, that's handled much higher up, in the Express layer. This is here to be imported as the styles weren't being part of our bundle.  I agree there could be a better way, but honestly I think we're going to use this component once we build it out anyway.

File: client/server/pages/index.js (hunk 3 above)
Comment (ChaosExAnima) [reply client/server/pages/index.js]: Agreed, let me see if I can resolve that.

Comment (ChaosExAnima) [reply client/server/pages/index.js]: Upon review, I actually think I'll defer. I think that passing in the `app` is messy, but I can't see a good way without combining the functions somehow.

I'll merge as-is, if that's okay :)

**Pull request merged**